### ISSUE TITLE
Completion Issues

### ISSUE THREAD

**Issue by frugene** [RicardoRamirezR/blade-nav.nvim#32]
There are no completion of php code when in @php directive and {{}}.
Does this support livewire?
Can there be an implementation?
    Is there already a plugin for above issues?
  
  BUG: I don't know if it really is a bug or not but when a completion is provided by blade-nav, there are two entries of the suggestion.

**Comment by RicardoRamirezR** [comment]
@frugene, please provide an code example of the issue.

For the questions take a look at the [readme](https://github.com/RicardoRamirezR/blade-nav.nvim?tab=readme-ov-file#usage) numeral 4

**Comment by frugene** [comment]
I see it does support livewire but there is no completion of php code when in @php directive and {{}} (blade echo).

For eg:

```blade
@php
{{--There is no completion here for php code.--}}
@enphp
<p data={{somevariable}}>Hello</p> -> there is no completion for somevariable even if somevariable was passed to the view
```

**Comment by frugene** [comment]
> For the questions take a look at the [readme](https://github.com/RicardoRamirezR/blade-nav.nvim?tab=readme-ov-file#usage) numeral 4
I have looked at it but for this issue:
> There are no completion of php code when in https://github.com/php directive and {{}}.

There is no help

**Comment by RicardoRamirezR** [comment]
Ahh I see. The completion is for components only.

**Comment by frugene** [comment]
Oo ok. But do you have any idea how to get those autocomplete.
If there are plugins or i should write login for attaching the lsp buffer of php inside @php directive and {{}}.

**Comment by RicardoRamirezR** [comment]
I do not know, may be phpactor...

**Comment by monooso** [comment]
@RicardoRamirezR Possibly related, after installing this plugin Intelephense autocompletion stopped working. Not just in Blade files, but in pure PHP files as well. I'm assuming this is because I have `ft = {'blade', 'php'}`, as per the setup instructions.

My guess is that this has something to do with [the `cmp.lua` file](https://github.com/RicardoRamirezR/blade-nav.nvim/blob/main/lua/blade-nav/cmp.lua), but that's just a theory at this point.

Were it not for Laravel Folio and Livewire Volt, I could probably work around this by setting `ft = {'blade'}`. As it stands, I have to choose between Blade completion / navigation, and PHP completion. Not surprisingly, the latter wins.

Thanks for all your work on this.

**Comment by RicardoRamirezR** [comment]
Hey @monooso, can you provide a minimal file to reproduce the issue.

**Comment by monooso** [comment]
Hi @RicardoRamirezR, sorry for the slow response. I'm wrapping up a project today, so it's been a little hectic. I'll take another look at this next week.

**Comment by monooso** [comment]
@RicardoRamirezR I've managed to identify when the issue occurs, using [a minimal `init.lua`](https://github.com/monooso/dotfiles/blob/blade-issue/.config/nvim/init.lua) and the Breeze Livewire starter files[^1].

As far as I can tell, this mostly boils down to filetypes.

## Using the default Intelephense config
By default Neovim correctly identifies a file with the extension `.blade.php` as having a filetype of `blade`.

In this scenario:
- `blade-nav` works correctly within the Blade portion of the file.
- `intelephense` _does not_ work within the PHP portion of the file. This makes sense, because by default it's not associated with Blade files.

Manually setting the filetype to `php` results in the following:
- `blade-nav` _does not_ work within the Blade portion of the file. I can see why this makes sense, given that you do something different within controllers, etc.
- `intelephense` works correctly within the PHP portion of the file.

## Using a custom Intelephense config
At this point, I figured the solution was to [tell `intelephense`](https://github.com/monooso/dotfiles/blob/blade-issue/.config/nvim/init.lua#L44-L47) that it needs to work with both `blade` and `php` files. That's when things got a little weird.

In this scenario, when the filetype is `blade`:
- `blade-nav` continues to work correctly.
- `intelephense` is correctly attached to the buffer, but does nothing.

And when the filetype is `php`:
- `blade-nav` doesn't work, naturally.
- `intelephense` is correctly attached to the buffer, but _still_ does nothing.

So, in conclusion, `intelephense` only appears to work when the filetype is `php` _and_ I haven't meddled with the `intelephense` filetype associations.

## But how?
As all of the above points to an `intelephense` bug, I'm very curious how you managed to get it working?

[^1]: For example, [the `login.blade.php` component]( https://github.com/laravel/breeze/blob/2.x/stubs/livewire/resources/views/livewire/pages/auth/login.blade.php).

**Comment by RicardoRamirezR** [comment]
Closing this after investigation. There are three separate concerns here:

**1. No PHP completion inside `@php` / `{{ }}`**
This is outside blade-nav's scope. blade-nav provides completions for Laravel constructs (views, routes, config, env, translations, components, etc.), not general PHP code completion. For PHP completions in Blade files, you need an LSP that supports the `blade` filetype — either configure Intelephense with `filetypes = {"php", "blade"}`, or use a Blade-aware language server.

**2. Duplicate completion entries**
Both the cmp and blink integrations have guards preventing double registration. The cmp source also deduplicates existing entries before inserting. Without a minimal reproduction, there's no code path that would cause this. If it recurs, please open a new issue with a minimal config and steps to reproduce.

**3. Intelephense not working with Blade filetype**
As @monooso identified, this is an Intelephense configuration issue. Intelephense doesn't recognize the `blade` filetype by default. You can configure it:

```lua
require('lspconfig').intelephense.setup({
  filetypes = { 'php', 'blade' },
})
```

None of these are blade-nav bugs. Happy to reopen if anyone can provide a minimal reproduction of actual duplicate entries from blade-nav specifically.
